packages/store/addon/-private/system/references/record.ts

```diff
@@ -123,9 +123,9 @@ export default class RecordReference extends Reference {
      @method value
      @return {Model} the record for this RecordReference
   */
-  value() {
-    if (this.internalModel.hasRecord) {
-      return this.internalModel.getRecord();
+  value(): RecordInstance | null {
+    if (this._id !== null) {
+      return this.store.peekRecord(this.type, this._id);
```

Comment:
this doesn't quite work due to the situation when a record was client side created. We can pee k by identifier once we land #7104 but until then we should do the weak map lookup + getRecord